viper/parser.py

```diff
@@ -668,12 +672,16 @@ def parse_expr(expr, context):
                                       typ=BaseType('decimal', new_unit), pos=getpos(expr))
         else:
             raise Exception("Unsupported binop: %r" % expr.op)
+
         if o.typ.typ == 'num':
             return LLLnode.from_list(['clamp', ['mload', MINNUM_POS], o, ['mload', MAXNUM_POS]], typ=o.typ, pos=getpos(expr))
         elif o.typ.typ == 'decimal':
             return LLLnode.from_list(['clamp', ['mload', MINDECIMAL_POS], o, ['mload', MAXDECIMAL_POS]], typ=o.typ, pos=getpos(expr))
+        elif o.typ.typ == 'num256':
+            return o
```

Comment:
We can return `o` directly here because there is no need for an overflow check? Is that correct?

Reply:
@fubuloubu yes that was my understanding, ie. the full 32byte has been occupied by a single variable, therefore can't overflow. But I could be wrong ? I worked from the LLL output of `num256_` functions they definitely don't have any sort of clamping - If we do require it, let me know and I will add it :)

Reply:
@jacqueswww I believe you should add clamping so that this test case results in `TransactionFailed` (right now it passes).  Clamping here is necessary because it checks user input (which can result in a number more than 2**256-1).
""" 

    def test_num_256_overflow():
        num256_code = """
    def _num256_mul(x: num256, y: num256) -> num256:
        return num256_mul(x, y)

    c = get_contract_with_gas_estimation(num256_code)
    x = 2 ** 256 - 1
    assert c._num256_mul(x, x) == 1

Reply:
Perfect @DavidKnott!

Reply:
Great stuff thanks for the info, will add soonest ;)